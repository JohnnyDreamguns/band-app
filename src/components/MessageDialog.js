import React from 'react';

const MessageDialog = ({ messages }) => (
  <React.Fragment>
    {messages.error && (
      <div className={'notify-box error'}>{messages.error}</div>
    )}
    {messages.info && <div className={'notify-box info'}>{messages.info}</div>}
  </React.Fragment>
);

export default MessageDialog;
