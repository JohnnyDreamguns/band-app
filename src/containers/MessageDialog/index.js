import React from 'react';
import { connect } from 'react-redux';
import { selectMessages } from './selectors';

const MessageDialog = ({ messages }) => (
  <React.Fragment>
    {messages.error && (
      <div className={'notify-box error'}>{messages.error}</div>
    )}
    {messages.info && <div className={'notify-box info'}>{messages.info}</div>}
  </React.Fragment>
);

export const mapStateToProps = state => {
  return {
    messages: selectMessages(state)
  };
};

export default connect(mapStateToProps)(MessageDialog);
