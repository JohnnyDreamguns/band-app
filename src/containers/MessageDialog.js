import React from 'react';
import MessageDialog from '../components/MessageDialog';
import { useMessages } from '../hooks/useMessages';

const MessageDialogContainer = () => {
  const { messages } = useMessages();

  const props = {
    messages
  };

  return <MessageDialog {...props} />;
};

export default MessageDialogContainer;
