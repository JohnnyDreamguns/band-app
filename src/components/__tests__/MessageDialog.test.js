import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MessageDialog from '../MessageDialog';

const renderer = new ShallowRenderer();

const props = {
  messages: {
    error: 'There was an error',
    info: 'This is for info'
  }
};

describe('<MessageDialog />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<MessageDialog {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
