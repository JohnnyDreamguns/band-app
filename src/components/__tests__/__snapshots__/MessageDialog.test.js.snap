// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageDialog /> should render and match the snapshot 1`] = `
<React.Fragment>
  <div
    className="notify-box error"
  >
    There was an error
  </div>
  <div
    className="notify-box info"
  >
    This is for info
  </div>
</React.Fragment>
`;
